### ISSUE TITLE
version 1.3.0 broken

### ISSUE THREAD

**Issue by markokocic** [liquidz/antq#125]
Steps to reproduce:

````
> cd ~/.clojure/
> cat deps.edn
{
 :aliases
 {:outdated
  {:extra-deps {com.github.liquidz/antq {:mvn/version "1.3.0"}} :main-opts ["-m" "antq.core"]}}
 }

> clojure -M:outdated
Execution error (PatternSyntaxException) at java.util.regex.Pattern/error (Pattern.java:2027).
Unexpected internal error near index 1
\

Full report at:
C:\Users\Marko.Kocic\scoop\apps\msys2\2021-11-30\tmp\clojure-13384489796296388982.edn

> cat /tmp/clojure-13384489796296388982.edn
{:clojure.main/message
 "Execution error (PatternSyntaxException) at java.util.regex.Pattern/error (Pattern.java:2027).\r\nUnexpected internal error near index 1\r\n\\\r\n",
 :clojure.main/triage
 {:clojure.error/class java.util.regex.PatternSyntaxException,
  :clojure.error/line 2027,
  :clojure.error/cause "Unexpected internal error near index 1\r\n\\",
  :clojure.error/symbol java.util.regex.Pattern/error,
  :clojure.error/source "Pattern.java",
  :clojure.error/phase :execution},
 :clojure.main/trace
 {:via
  [{:type java.util.regex.PatternSyntaxException,
    :message "Unexpected internal error near index 1\r\n\\",
    :at [java.util.regex.Pattern error "Pattern.java" 2027]}],
  :trace
  [[java.util.regex.Pattern error "Pattern.java" 2027]
   [java.util.regex.Pattern compile "Pattern.java" 1788]
   [java.util.regex.Pattern <init> "Pattern.java" 1428]
   [java.util.regex.Pattern compile "Pattern.java" 1068]
   [java.lang.String split "String.java" 2317]
   [java.lang.String split "String.java" 2364]
   [antq.util.dep$normalize_path invokeStatic "dep.clj" 37]
   [antq.util.dep$normalize_path invoke "dep.clj" 34]
   [antq.dep.clojure$load_deps invokeStatic "clojure.clj" 145]
   [antq.dep.clojure$load_deps invoke "clojure.clj" 141]
   [antq.dep.clojure$load_deps invokeStatic "clojure.clj" 143]
   [antq.dep.clojure$load_deps invoke "clojure.clj" 141]
   [antq.core$fetch_deps$fn__8581 invoke "core.clj" 200]
   [clojure.core$map$fn__5884 invoke "core.clj" 2759]
   [clojure.lang.LazySeq sval "LazySeq.java" 42]
   [clojure.lang.LazySeq seq "LazySeq.java" 51]
   [clojure.lang.RT seq "RT.java" 535]
   [clojure.core$seq__5419 invokeStatic "core.clj" 139]
   [clojure.core$apply invokeStatic "core.clj" 662]
   [clojure.core$mapcat invokeStatic "core.clj" 2787]
   [clojure.core$mapcat doInvoke "core.clj" 2787]
   [clojure.lang.RestFn invoke "RestFn.java" 423]
   [antq.core$fetch_deps invokeStatic "core.clj" 198]
   [antq.core$fetch_deps invoke "core.clj" 195]
   [antq.core$main_STAR_ invokeStatic "core.clj" 246]
   [antq.core$main_STAR_ invoke "core.clj" 240]
   [antq.core$_main invokeStatic "core.clj" 270]
   [antq.core$_main doInvoke "core.clj" 266]
   [clojure.lang.RestFn invoke "RestFn.java" 397]
   [clojure.lang.AFn applyToHelper "AFn.java" 152]
   [clojure.lang.RestFn applyTo "RestFn.java" 132]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.main$main_opt invokeStatic "main.clj" 514]
   [clojure.main$main_opt invoke "main.clj" 510]
   [clojure.main$main invokeStatic "main.clj" 664]
   [clojure.main$main doInvoke "main.clj" 616]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.main main "main.java" 40]],
  :cause "Unexpected internal error near index 1\r\n\\"}}
>
````

**Comment by liquidz** [comment]
@markokocic Thanks for your reporting!

It seems Windows specific problem.
https://github.com/liquidz/antq/blob/6c0d2173c0d445ab0e39b7174bc55cd4acb5f652/src/antq/util/dep.clj#L34

I don't have WIndows environment, but I'll try to fix.

**Comment by markokocic** [comment]
I admit I'm not familiar with the reason behind `normalize-path` function, but is there a reason why not using `java.io.File/getCanonicalPath` instead of custom path parsing?

**Comment by liquidz** [comment]
@markokocic Oh, I missed `getCanonicalPath`! Thank!

But we want to return a relative path with `normalize-path` if possible.
Do you have a good way?

The purpose of `normalize-path` is to normalize the directory specified by `:local/root` in deps.edn.
Normalized paths are used to detect infinite loops and report results.

Anyway, I patched a workaround for now.
Could you try [dev](https://github.com/liquidz/antq/tree/dev) branch?

**Comment by markokocic** [comment]
> But we want to return a relative path with normalize-path if possible.
Do you have a good way?

In Java you can look at `java.nio.file/Path` [docs](https://docs.oracle.com/javase/8/docs/api/?java/nio/file/Path.html). Maybe `normalize`, `releativize` and `resolve` methods can help you.

> Anyway, I patched a workaround for now.
Could you try dev branch?

I tried dev branch and it seems to fix error.

**Comment by liquidz** [comment]
@markokocic Sorry for late reply.

> In Java you can look at java.nio.file/Path docs. Maybe normalize, releativize and resolve methods can help you.

Thanks so much for letting me know!

> I tried dev branch and it seems to fix error.

Great. Thanks.
I'll cut a new release.

**Comment by liquidz** [comment]
Just released v1.3.1 #128
